packages/core/src/api/apis/definitions/oauthrequest.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IconComponent } from '../../../icons';
+import { Observable } from '../../types';
+import { ApiRef } from '../ApiRef';
+
+export type OAuthScopes = {
```

Comment:
What do you think about getting rid of this type and have this API just work with strings and list of strings?

Reply:
As discussed on Discord, we'll probably revisit this once we get some auth implementations in. It may be that we can get rid of this interface altogether.